apps/bfd-model/bfd-model-rif-samples/dev/synthea-test-plan.md

```diff
@@ -0,0 +1,280 @@
+# Synthea Test Plan
+
+This document details various tests/procedures to ensure that each batch of synthetic data generated will not collide with previously generated data. 
+Synthetic data is generated with the external Synthea codebase. More information on this process can be found at: `apps/bfd-model/bfd-model-rif-samples/dev/design-sample-data-sets.md`.
+
+## Before Generating Data
+
+1. Each time you generate synthea data, you are also generating ranges of values for several beneficiary and claim data properties. Furthermore, this creates database constraints for subsequent generated synthea datasets, which will cause collisions if not set properly. The input is the set of start values that the data can have, i.e the lower bound of these constraints. Once the data is generated, there is an output file representing the end values of the data, or the upper bound. Before generating data, go to `apps/bfd-model/bfd-model-rif-samples/dev/synthea_releases/XX-XX-20XX/end_state.properties` for the latest end-state properties. See example below: 
+
+        exporter.bfd.clm_grp_id_start=-104410371
+        exporter.bfd.pde_id_start=-10000464592
+        exporter.bfd.carr_clm_cntl_num_start=-100518629
+        exporter.bfd.fi_doc_cntl_num_start=-100251672
+        exporter.bfd.hicn_start=T01030000A
+        exporter.bfd.bene_id_start=-10000000020000
+        exporter.bfd.clm_id_start=-10000003945779
+        exporter.bfd.mbi_start=1S00E00NA00
+
+2. With the end-state properties in mind, you need to check for two things before setting the next synthea property start values. First, if you increment each end-state property by 1, will the amount of data you are trying to generate e.g. 10,000, allow for a new range of values that has no overlap with previously generated synthea datasets? To answer this you need to run queries in PROD SBX. Run `psql -h {prod sbx db url} -d fhirdb -U bfduser -f apps/bfd-model/bfd-model-rif/src/main/resources/db/scripts/Query_synthetic_id_ranges.sql | tee Query_synthetic_id_rangesTest.out` to generate a series of outputs for various end state properties. An example for beneficiary ID is shown below:
```

Comment:
I like this explanation. I think it would be nice though to make the two things to check for (First, and Second) separate bullets underneath this number 2 section since they logically go together.